esmvaltool/cmorizers/data/formatters/datasets/esacci_landcover.py

```diff
@@ -0,0 +1,378 @@
+logger = logging.getLogger(__name__)
+
+# Enable the new split-attributes handling mode
+iris.FUTURE.save_split_attrs = True
+
+
+def extract_variable(raw_info):
+    """Extract the variable from the raw data file."""
+    cube_list = iris.load(raw_info['file'], raw_info['name'])
+    if not cube_list:
+        logger.warning("No cubes found for {raw_info['name']} in file "
+                       "{raw_info['file']}")
+    else:
+        logger.info("Extracted cubes: %s", cube_list)
+    return cube_list
+
+
+def average_block(data, block_size):
+    """Average the data within each block of size block_size."""
+    shape = data.shape
+    reshaped_data = data.reshape(shape[0], shape[1] // block_size,
+                                 block_size, shape[2] // block_size,
+                                 block_size)
+    averaged_data = reshaped_data.mean(axis=(2, 4))
+    return averaged_data
+
+
+def regrid_iris(cube):
+    """Regrid the cubes using block averaging."""
+    logger.info("Regridding using block averaging")
+
+    block_size = 100
+
+    combined_data = average_block(cube.data, block_size)
+
+    # Define target latitude and longitude ranges
+    target_lats = np.linspace(90 - 0.5 * (180 / combined_data.shape[1]),
+                              -90 + 0.5 * (180 / combined_data.shape[1]),
+                              combined_data.shape[1])
+    target_lons = np.linspace(-180 + 0.5 * (360 / combined_data.shape[2]),
+                              180 - 0.5 * (360 / combined_data.shape[2]),
+                              combined_data.shape[2])
+
+    # Flip the latitude points and data
+    combined_data = combined_data[:, ::-1, :]
+
+    lat_bounds = calculate_bounds(target_lats[::-1])
+    lon_bounds = calculate_bounds(target_lons)
+
+    combined_cube = iris.cube.Cube(combined_data,
+                                   dim_coords_and_dims=[
+                                       (cube.coord('time'), 0),
+                                       (iris.coords.DimCoord(
+                                           target_lats[::-1],
+                                           standard_name='latitude',
+                                           units='degrees',
+                                           bounds=lat_bounds), 1),
+                                       (iris.coords.DimCoord(
+                                           target_lons,
+                                           standard_name='longitude',
+                                           units='degrees',
+                                           bounds=lon_bounds), 2)])
+
+    return combined_cube
+
+
+def calculate_bounds(points):
+    """Calculate bounds for a set of points."""
+    bounds = np.zeros((len(points), 2))
+    bounds[1:, 0] = (points[:-1] + points[1:]) / 2
+    bounds[:-1, 1] = bounds[1:, 0]
+    bounds[0, 0] = points[0] - (bounds[1, 0] - points[0])
+    bounds[-1, 1] = points[-1] + (points[-1] - bounds[-2, 1])
+    return bounds
```

Comment:
```suggestion
    combined_cube = iris.cube.Cube(combined_data,
                                   dim_coords_and_dims=[
                                       (cube.coord('time'), 0),
                                       (iris.coords.DimCoord(
                                           target_lats[::-1],
                                           standard_name='latitude',
                                           units='degrees', 1),
                                       (iris.coords.DimCoord(
                                           target_lons,
                                           standard_name='longitude',
                                           units='degrees', 2)])

    combined_cube.coord('latitude').guess_bounds()
    combined_cube.coord('longitude').guess_bounds()
    return combined_cube
```

You don't need a custom solution for calculating the bounds